### DISCUSSION TITLE
Feature request (new System Handle): regarding using ZeroMQ as one of the endpoint

### DISCUSSION THREAD

**Discussion by lalat-das** [eProsima/Integration-Service#138] [Ideas]
Hi 
I was exploring the SOSS & ROSIS github, it is a very promising idea to implement for connecting the ROS2 with rest of the world.
I was mostly interested to see if there is proof of concept level  implementation or design discussion around using ROS2 and ZEROMQ as two endpoint using SOSS infrastructure ? Any pointers would be highly appreciable. Or any obvious issue/bottlenecks we foresee for connecting SOSS-ROS2 and SOSS-ZMQ (if any)

**Comment by mxgrey** [comment]
ZeroMQ is definitely a middleware that we imagine would be a great candidate for SOSS integration.

We haven't done any work on it, and it's not on our roadmap, but that's only because our resources are limited and we haven't had any need to interoperate with a ZeroMQ platform yet.

If it's something you're interested in contributing to the community, I can think of two ways to go about it:

1. You could make your own repo to host `soss-zeromq` similar to [`soss-ros1`](https://github.com/osrf/soss-ros1).
2. You could make a PR that adds it to this repo, and we can run it through our code review process.

I think I would strongly recommend option (1) because I'd be reluctant to add more dependencies to this repo (and in fact I think the `soss-ros2` package in this repo really belongs in its own). We might be able to offer some code review effort, even if you choose option (1). Just follow up with us if/when you think you have something ready for review.
